### DISCUSSION TITLE
Can you re-run getStaticPaths on some interval?

### DISCUSSION THREAD

**Discussion by stevenschmatz** [vercel/next.js#19723] [Help] [answered]
We are building a web application for guest check-in forms. We have URLs of the format `booking/[id].js`, and the list of booking IDs changes regularly.  Each of these paths will only be accessed once typically, but we want to have these pages pre-rendered ahead of time for each guest.

Ideally, on a regular interval (say 24hrs) we would want to re-run `getStaticPaths` to get a new list of check-in eligible booking IDs for that day. 

Is there any way to do this without writing a job to redeploy every 24 hours?

**Comment by jamesmosier** [comment, marked as answer]
There is no built-in way to do this with Next on a time interval, but you could use [revalidate](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) to re-generate your page on an interval. This would work by rebuilding it in the background whenever that interval is hit and that would update your list of bookings.

**Reply by stevenschmatz** [reply]
Gotcha @jamesmosier, thanks! I didn't know that `revalidate` also caused `getStaticPaths` to re-run on that interval

**Reply by dylanmrowe** [reply]
@jamesmosier Does revalidate really cause getStaticPaths to re-run? I don't see anything about that in the docs. Only getStaticProps

We have a similar use case where an author is creating new content under a new URL. Once they're done, the URL path is added to the list that getStaticPaths retrieves. We want getStaticPaths to pull that updated list without re-building and re-deploying.

**Reply by dylanmrowe** [reply]
Nevermind, found it. Looks like using 'fallback:true' does what we want. Sweet!

https://nextjs.org/docs/basic-features/data-fetching#fallback-true
